contracts/core/merge/Eater.sol

```diff
@@ -22,15 +30,24 @@ abstract contract Eater is IWeweReceiver, ReentrancyGuard, Pausable, Ownable {
         }
     }
 
-    function _merge(uint256 amount, address token, address from) internal whenNotPaused {
-        uint256 weweToTransfer = (amount * _rate) / 100;
+    function _merge(uint256 amount, address token, address from) internal {
+        uint256 vestedAmount = vestings[from].amount;
+        uint256 weweToTransfer = vestedAmount + (amount * _rate) / 100;
```

Comment:
We cannot use the from parameter to get previous vestedAmount if we do so we would be allowing the user to put in a from of another user and end up claiming more than his due, draining the contract

Reply:
As discussed, this is an internal.